datamonster_api/lib/datamonster.py

```diff
@@ -199,61 +237,50 @@ def get_data(
             if aggregation.period is not None:
                 params["aggregation"] = aggregation.period
 
-        url = "{}/{}/data?{}".format(
-            self.datasource_path,
-            datasource.id,
-            six.moves.urllib.parse.urlencode(params),
-        )
         headers = {"Accept": "avro/binary"}
-        resp = self.client.get(url, headers)
+        url = self._get_rawdata_path(datasource.id, params)
+        resp = self.client.get(url, headers, stream=True)
+        split_columns = datasource.get_details()["splitColumns"]
+        return self._avro_to_df(resp.content, split_columns)
 
-        return self._avro_to_df(resp)
-
-    def get_datasource_details(self, datasource_id):
-        """Get details (metadata) for the given datasource
+    def _avro_to_df(self, avro_buffer, split_columns):
+        """Read an avro structure into a dataframe
 
-        :param datasource_id: The ID of the datasource for which we get the details
-        :return: dictionary object with the datasource details
+        Transforms dates and columns to a stanard and agreed upon format
         """
-        path = self._get_datasource_path(datasource_id)
-        return self.client.get(path)
-
-    def _get_datasource_path(self, datasource_id):
-        return "{}/{}".format(self.datasource_path, datasource_id)
-
-    def _datasource_result_to_object(self, datasource, has_details=False):
-        ds_inst = Datasource(
-            datasource["id"],
-            datasource["name"],
-            datasource["category"],
-            datasource["uri"],
-            self,
-        )
-
-        if has_details:
-            ds_inst.set_details(datasource)
 
-        return ds_inst
+        def parse_row(row, data_column):
+            return {
+                "value": row[data_column],
+                "start_date": row["period_start"],
```

Comment:
These shouldn't be hard coded. Technically, the period_start and period_end columns can have different names is specified as such in the avro schema.